### PR TITLE
Support case-insensitive renaming

### PR DESCRIPTION
**Pull request by crisptrutski**
Refs #34 

Not much to say here. Seems too simple, perhaps I'm missing some interesting edge cases?

### CODE REVIEW

File: test/macaw/core_test.clj [hunk 1]
```diff
@@ -264,6 +264,14 @@
   (is (thrown? Exception #"Unknown rename"
                (m/replace-names "SELECT 1" {:tables {{:schema "public" :table "a"} "aa"}}))))
 
+(deftest case-insensitive-test
+  ;; In future we might try to be smarter and preserve case.
+  (is (= "SELECT cats.meow FROM cats"
+         (m/replace-names "SELECT DOGS.BaRk FROM dOGS"
+                          {:tables  {{:table "dogs"} "cats"}
+                           :columns {{:table "dogs" :column "bark"} "meow"}}
+                          :case-insensitive? true))))
```
Comment (piranha) [inline test/macaw/core_test.clj:273]: Ehhh IDK I dislike "naked" options, they are kind of weird and also you have to align them manually. But not enough not to approve this.

As for missing some cases: we'll see I guess? :)

**Review by piranha — APPROVED** [review]

File: test/macaw/core_test.clj (hunk 1 above)
Comment (crisptrutski) [reply test/macaw/core_test.clj:273]: Cool, I'll keep this in mind for future. 

I think I also prefer just using maps, but picked up this style from code I've seen in the main repo.

File: test/macaw/core_test.clj [hunk 2]
```diff
@@ -264,6 +264,14 @@
   (is (thrown? Exception #"Unknown rename"
                (m/replace-names "SELECT 1" {:tables {{:schema "public" :table "a"} "aa"}}))))
 
+(deftest case-insensitive-test
+  ;; In future we might try to be smarter and preserve case.
```
Comment (tsmacdonald) [inline test/macaw/core_test.clj:268]: I'm not sure that's necessary. People can put in whatever case they want as the new name and it seems right and meet to just respect that

Comment (crisptrutski) [reply test/macaw/core_test.clj:268]: I came into this as a hard liner for preserving formatting choices, but yeah being opinionated and propagating the field's default spelling makes more sense. I'll nix the comment next time I come through here.
